src/lib/crypto/kimchi_backend/gadgets/xor.ml

```diff
@@ -0,0 +1,333 @@
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    (input1 : Circuit.Field.t) (input2 : Circuit.Field.t) : Circuit.Field.t =
+  bxor (module Circuit) input1 input2 64 4
+
+let%test_unit "xor gadget" =
+  (* Import the gadget test runner *)
+  let open Kimchi_gadgets_test_runner in
+  (* Initialize the SRS cache. *)
+  let () =
+    try Kimchi_pasta.Vesta_based_plonk.Keypair.set_urs_info [] with _ -> ()
+  in
+
+  (* Helper to test Xor gadget
+     *   Inputs operands and expected output: left_input xor right_input
+     *   Returns true if constraints are satisfied, false otherwise.
+  *)
+  let test_xor left_input right_input output_xor length : unit =
+    let _proof_keypair, _proof =
+      Runner.generate_and_verify_proof (fun () ->
+          let open Runner.Impl in
+          (* Set up snarky variables for inputs and output *)
+          let left_input =
+            exists Field.typ ~compute:(fun () ->
+                Field.Constant.of_string left_input )
+          in
+          let right_input =
+            exists Field.typ ~compute:(fun () ->
+                Field.Constant.of_string right_input )
+          in
+          let output_xor =
+            exists Field.typ ~compute:(fun () ->
+                Field.Constant.of_string output_xor )
+          in
+          (* Use the xor gate gadget *)
+          let result =
+            bxor (module Runner.Impl) left_input right_input length 4
+          in
+          Field.Assert.equal output_xor result ;
+          (* Pad with a "dummy" constraint b/c Kimchi requires at least 2 *)
+          Boolean.Assert.is_true (Field.equal result result) )
+    in
+    ()
+  in
+
+  let test_2xor left1 right1 output1 left2 right2 output2 length : unit =
+    let _proof_keypair, _proof =
+      Runner.generate_and_verify_proof (fun () ->
+          let open Runner.Impl in
+          (* Set up snarky variables for inputs and output *)
+          let left1 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) left1
+          in
+          let right1 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) right1
+          in
+          let output1 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) output1
+          in
+          let left2 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) left2
+          in
+          let right2 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) right2
+          in
+          let output2 =
+            Common.as_prover_cvar_field_of_base10 (module Runner.Impl) output2
+          in
+          (* Use the xor gate gadget *)
+          let result1 = bxor (module Runner.Impl) left1 right1 length 4 in
+          let result2 = bxor (module Runner.Impl) left2 right2 length 4 in
+          Field.Assert.equal output1 result1 ;
+          Field.Assert.equal output2 result2 ;
+          Boolean.Assert.is_true (Field.equal result1 result1) )
```

Comment:
I think you don't need this padding since your xor is always more than one gate.

Reply:
okay!